controllers/prep_handlers.go

```diff
@@ -214,6 +231,31 @@ func checkSeedImageProxyCompatibility(seedHasProxy, hasProxy bool) error {
 	return nil
 }
 
+// checkSeedImageAdditionalTrustBundleCompatibility checks for proxy
+// configuration compatibility of the seed image vs the current cluster. If the
+// seed image has a proxy and the cluster being upgraded doesn't, we cannot
+// proceed as recert does not support proxy rename under those conditions.
+// Similarly, we cannot proceed if the cluster being upgraded has a proxy but
+// the seed image doesn't.
+func checkSeedImageAdditionalTrustBundleCompatibility(seedAdditionalTrustBundle seedclusterinfo.AdditionalTrustBundle, hasUserCaBundle bool, proxyConfigmapName string) error {
+	if seedAdditionalTrustBundle.HasUserCaBundle && !hasUserCaBundle {
+		return fmt.Errorf("seed image has an %s/%s configmap but the cluster being upgraded does not, this combination is not supported",
+			common.OpenshiftConfigNamespace, common.ClusterAdditionalTrustBundleName)
+	}
+
+	if !seedAdditionalTrustBundle.HasUserCaBundle && hasUserCaBundle {
+		return fmt.Errorf("seed image does not have an %s/%s configmap but the cluster being upgraded does, this combination is not supported",
+			common.OpenshiftConfigNamespace, common.ClusterAdditionalTrustBundleName)
+	}
+
+	if seedAdditionalTrustBundle.ProxyConfigmapName != proxyConfigmapName {
```

Comment:
Hmm, i wonder why we should not support it? Can't recert just get both names?

Reply:
There's already a lot of different cases we must support, supporting also another case where the name of the configmap holding the proxy trust bundle is different between the seed and the new cluster is just too much work (recert would have to delete the new configmap, create a new configmap with the correct name, and change the proxy reference to point to the new configmap) for something that's completely unnecessary. If we'll have users that have different names for this configmap across multiple clusters which they want to update with the same seed, we can consider supporting this, until then, I don't see the point.